framework/src/wifi_manager/wifi_profile.c

```diff
@@ -232,18 +238,27 @@ wifi_utils_result_e wifi_profile_write(wifi_manager_ap_config_s *config)
 
 wifi_utils_result_e wifi_profile_read(wifi_manager_ap_config_s *config)
 {
-	char buf[256] = {0,};
+	char buf[128] = {0,};
 	int ret = -1;
 #ifdef CONFIG_WIFI_PROFILE_SECURESTORAGE
-	unsigned int readlen = 256;
+	/////////////////////////////////////////////////////////////////////////////////////////
+	//
+	// max size:
+	// ssid(31) + tab(1) + ssid_length(2) + tab(1) + ap_type(1) + tab(1) +
+	// passphrase(63) + tab(1) + passphrase length(2) + tab(1) + crypto type(1) == 105
+	// file size of wifi.conf should not exceed 107(+EOF)
+	// ssid_length and passphrase length are stored as character.
+	//
+	/////////////////////////////////////////////////////////////////////////////////////////
+	unsigned int readlen = 128;
 	ret = see_read_secure_storage((unsigned char *)buf, &readlen, WIFI_PROFILE_SS_INDEX);
 	if (ret != SEE_OK) {
 		return WIFI_UTILS_FILE_ERROR;
-	} else if (readlen == 1) {
+	} else if (readlen == 1 || readlen > 128) {
```

Comment:
What does readlen == 1 mean?

Reply:
I think that it is not needed anymore. as I remember, readlen == 1 points that read fail. but it's not valid anymore. so I'll remove it